fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java

```diff
@@ -224,10 +224,13 @@ public String toString() {
     }
 
     public void prepare(OptExpression queryOptExpression, MvRewriteStrategy strategy) {
+        SessionVariable sessionVariable = connectContext.getSessionVariable();
         // MV Rewrite will be used when cbo is enabled.
-        if (context.getOptimizerConfig().isRuleBased()) {
+        if (context.getOptimizerConfig().isRuleBased() || sessionVariable.isDisableMaterializedViewRewrite() ||
+                !sessionVariable.isEnableMaterializedViewRewrite()) {
             return;
```

Comment:
`sessionVariable.isDisableMaterializedViewRewrite()` and `sessionVariable.isEnableMaterializedViewRewrite()` are different which is a bit of confused.

Reply:
You are right.  They are used for different scenes. We can refactor it later.